packages/near-membrane-base/src/__tests__/intrinsics.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { VirtualEnvironment } from '../../types';
+import { isIntrinsicGlobalName, linkIntrinsics } from '../intrinsics';
+
+const intrinsicNames = [
+    'Infinity',
+    'NaN',
+    'undefined',
+    'eval', // dangerous
+    'isFinite',
+    'isNaN',
+    'parseFloat',
+    'parseInt',
+    'decodeURI',
+    'decodeURIComponent',
+    'encodeURI',
+    'encodeURIComponent',
+    'AggregateError',
+    'Array',
+    'ArrayBuffer',
+    'BigInt',
+    'BigInt64Array',
+    'BigUint64Array',
+    'Boolean',
+    'DataView',
+    'Error', // Unstable
+    'EvalError',
+    'FinalizationRegistry',
+    'Float32Array',
+    'Float64Array',
+    'Function', // dangerous
+    'Int8Array',
+    'Int16Array',
+    'Int32Array',
+    'Number',
+    'Object',
+    'Proxy', // Unstable
+    'RangeError',
+    'ReferenceError',
+    'RegExp', // Unstable
+    'SharedArrayBuffer',
+    'String',
+    'Symbol',
+    'SyntaxError',
+    'TypeError',
+    'Uint8Array',
+    'Uint8ClampedArray',
+    'Uint16Array',
+    'Uint32Array',
+    'URIError',
+    'WeakRef',
+    'Atomics',
+    'JSON',
+    'Math',
+    'Reflect',
+    'escape',
+    'unescape',
+];
+const remappedIntrinsicNames = [
+    'Date', // Unstable & Remapped
+    'Map', // Remapped
+    'Promise', // Remapped
+    'Set', // Remapped
+    'WeakMap', // Remapped
+    'WeakSet', // Remapped
+    'Intl', // Unstable & Remapped
+];
+
+describe('isIntrinsicGlobalName()', () => {
+    it('should return false foe ES global names that are remapped', () => {
+        expect.assertions(remappedIntrinsicNames.length);
+        remappedIntrinsicNames.forEach((remappedIntrinsicName) => {
+            expect(isIntrinsicGlobalName(remappedIntrinsicName)).toBe(false);
+        });
+    });
+    it('should return true for all non-remapped ES global names', () => {
+        expect.assertions(intrinsicNames.length);
+        intrinsicNames.forEach((intrinsicName) => {
+            expect(isIntrinsicGlobalName(intrinsicName)).toBe(true);
+        });
+    });
+});
+
+describe('linkIntrinsics()', () => {
+    it('should attempt to link reflective intrinsics that are in the blue realm', () => {
+        // expect.assertions(2);
+        const fakeEnv = ({
+            setRefMapEntries: jest.fn(),
+        } as unknown) as VirtualEnvironment;
+        expect(() => {
+            linkIntrinsics(fakeEnv, globalThis, {} as typeof globalThis);
+            // We _want_ an error, because we know that redGlobalThis is
+            // empty and there for will have nothing to link to; neither
+            // will that nothing have a prototype to link to!
+        }).toThrow(/Cannot read property 'prototype' of undefined/);
```

Comment:
☝️ If that error works cross-browser rad. Else narrowing it to a TypeError or something is ok.

Reply:
I think this is a `ReferenceError`, but you're right about the error message itself. I'll update this.